### PR TITLE
Added interaction popup, restructured and refactored

### PR DESCRIPTION
**Pull request by CalebDevonport**
Added interaction popup, restructured folders and scenes, added character to prefabs, refactored some scripts, added geralds keys sprites

### CODE REVIEW

File: Assets/Scripts/Player/InteractionPopupController.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private void Start()
+    {
+        _animator = gameObject.GetComponent<Animator>();
+    }
+
+    public void SetActive()
+    {
+        if(!gameObject.activeSelf)
+        {
+            gameObject.SetActive(true);
```
Comment (ryanlaycock) [inline Assets/Scripts/Player/InteractionPopupController.cs]: Does this need to be in the if?

File: Assets/Scripts/Player/PlayerInteractionController.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            }
+        }
+    }
+
+    public void CheckForInteractions()
+    {
+        _hitColliders = Physics2D.OverlapCircleAll(transform.position, DetectionRadius, _interactionLayers);
+        if (_hitColliders.Any())
+        {
+            _interactionPopupController.SetActive();
```
Comment (ryanlaycock) [inline Assets/Scripts/Player/PlayerInteractionController.cs]: Does this call the SetActive method on the _interactionPopupController or the method on the gameObject? Maybe renaming the function to one that isn't already used would make it a bit clearer

**Review by ryanlaycock — APPROVED** [review]
Looks good, minor comments but its a POC so who cares :)

File: Assets/Scripts/Player/InteractionPopupController.cs (hunk 1 above)
Comment (CalebDevonport) [reply Assets/Scripts/Player/InteractionPopupController.cs]: I'm not actually sure, I just didn't think there was much point repeatedly calling the method if no need but I guess it's idempotent so doesn't really matter.

File: Assets/Scripts/Player/PlayerInteractionController.cs (hunk 2 above)
Comment (CalebDevonport) [reply Assets/Scripts/Player/PlayerInteractionController.cs]: Good shout maladdo

**Comment by CalebDevonport** [comment]
> Looks good, minor comments but its a POC so who cares :)

Who cares? I care 💪

**Pull request merged**